### ISSUE TITLE
GeoLite2-ASN.mmdb in Docker container is not up-to-date

### ISSUE THREAD

**Issue by laurentgoudet** [crowdsecurity/crowdsec#3281]
### What happened?

While I was investigating while a custom AS-based rule was not triggering for some of our traffic, I have found out that the version of `GeoLite2-ASN.mmdb` in the latest `crowdsecurity/crowdsec` container (`v1.6.3` at the moment) is older than the one available from MaxMind or the CrowdSec hub, causing those IPs to be missing in the old DB.

### What did you expect to happen?

`GeoLite2-ASN.mmdb` (and `GeoLite2-City.mmdb`) should be kept up-to-date for incoming traffic to be properly tagged by `geoip-enrich`.

### How can we reproduce it (as minimally and precisely as possible)?

```
$ docker cp $(docker create docker.io/crowdsecurity/crowdsec):/staging/var/lib/crowdsec/data/GeoLite2-ASN.mmdb GeoLite2-ASN-docker.mmdb
Successfully copied 8.41MB to /Users/laurent/Freelancer/temp-maxmind-dbs/GeoLite2-ASN-docker.mmdb
$ mmdblookup -f GeoLite2-ASN-docker.mmdb --verbose -i 57.141.0.13

  Database metadata
    Node count:    1018111
    Record size:   24 bits
    IP version:    IPv6
    Binary format: 2.0
    Build epoch:   1709648914 (2024-03-05 14:28:34 UTC)
    Type:          GeoLite2-ASN
    Languages:     en
    Description:
      en:   GeoLite2 ASN database


  Record prefix length: 112

  Could not find an entry for this IP address (57.141.0.13)
```

As seen above DB version in the latest image is `2024-03-05`, and misses some IPs. Current version from https://hub-data.crowdsec.net is `2024-09-27` and contains those IPs:

```
$ wget -q -O GeoLite2-ASN-web.mmdb https://hub-data.crowdsec.net/mmdb_update/GeoLite2-ASN.mmdb
$ mmdblookup -f GeoLite2-ASN-web.mmdb --verbose -i 57.141.0.13

  Database metadata
    Node count:    1055590
    Record size:   24 bits
    IP version:    IPv6
    Binary format: 2.0
    Build epoch:   1727424913 (2024-09-27 08:15:13 UTC)
    Type:          GeoLite2-ASN
    Languages:     en
    Description:
      en:   GeoLite2 ASN database


  Record prefix length: 120

  {
    "autonomous_system_number":
      32934 <uint32>
    "autonomous_system_organization":
      "FACEBOOK" <utf8_string>
  }
```

Note that https://hub-data.crowdsec.net/mmdb_update does not seem to be kept up-to-date that frequently either, compared to the version available directly from Maxmind:

```
$ wget -q -O GeoLite2-ASN-maxmind.tar.gz --content-disposition --user=<redacted> --password=<redacted> https://download.maxmind.com/geoip/databases/GeoLite2-ASN/download\?suffix\=tar.gz
$ tar -xzvf GeoLite2-ASN-maxmind.tar.gz
x GeoLite2-ASN_20241010/
x GeoLite2-ASN_20241010/COPYRIGHT.txt
x GeoLite2-ASN_20241010/GeoLite2-ASN.mmdb
x GeoLite2-ASN_20241010/LICENSE.txt
$ mmdblookup -f GeoLite2-ASN_20241010/GeoLite2-ASN.mmdb --verbose -i 57.141.0.13

  Database metadata
    Node count:    1057049
    Record size:   24 bits
    IP version:    IPv6
    Binary format: 2.0
    Build epoch:   1728548140 (2024-10-10 08:15:40 UTC)
    Type:          GeoLite2-ASN
    Languages:     en
    Description:
      en:   GeoLite2 ASN database


  Record prefix length: 120

  {
    "autonomous_system_number":
      32934 <uint32>
    "autonomous_system_organization":
      "FACEBOOK" <utf8_string>
  }
```

### Crowdsec version

<details>

```console
$ cscli version
version: v1.6.3-4851945a
Codename: alphaga
BuildDate: 2024-09-12_09:39:08
GoVersion: 1.22.6
Platform: docker
libre2: C++
User-Agent: crowdsec/v1.6.3-4851945a-docker
Constraint_parser: >= 1.0, <= 3.0
Constraint_scenario: >= 1.0, <= 3.0
Constraint_api: v1
Constraint_acquis: >= 1.0, < 2.0
```

</details>

**Comment by blotus** [comment]
Hello,

The MMDBs in `/staging` are just there to provide a "default" database, and the updated one should be downloaded from the hub automatically on start *if* `/var/lib/crowdsec/data/` is mounted as a volume (we do not update anything by design if the hub data folder is not a volume, as it can waste *a lot* of bandwidth if the user has misconfigured something and the container restarts in a loop).

If it's mounted in a volume and you do not get the version from the hub, then we have a bug.

The MMDBs are updated in the hub every 30 days (we could probably lower this a bit, but again, the idea is to avoid sending TBs worth of data every day).

You should be able to also copy your own MMDB files in `/var/lib/crowdsec/data,` and they should not be overwritten if they are more recent that the one provided in the hub (just make sure to use the same names, and the proper DB type for each file)

**Comment by laurentgoudet** [comment]
> The MMDBs in `/staging` are just there to provide a "default" database, and the updated one should be downloaded from the hub automatically on start _if_ `/var/lib/crowdsec/data/` is mounted as a volume (we do not update anything by design if the hub data folder is not a volume, as it can waste _a lot_ of bandwidth if the user has misconfigured something and the container restarts in a loop).

Oh right I see. Yeah I was mounting individual config files onto `/var/lib/crowdsec/data/` but I was not aware that the upgrade behavior was driven but mounting the whole folder; I've just tested that & I can confirm that this indeed works, the new file from the hub do gets pulled.
 
> The MMDBs are updated in the hub every 30 days (we could probably lower this a bit, but again, the idea is to avoid sending TBs worth of data every day).

Right make sense, thanks for the clarification.

> You should be able to also copy your own MMDB files in `/var/lib/crowdsec/data,` and they should not be overwritten if they are more recent that the one provided in the hub (just make sure to use the same names, and the proper DB type for each file)

That's what I was indeed doing as a workaround.

---

I guess I was expecting the base DB file in the Docker image to be updated at each release, but since it does get pulled from the Web once `/var/lib/crowdsec/data/` is mounted everything is working as designed.

**Comment by blotus** [comment]
> I guess I was expecting the base DB file in the Docker image to be updated at each release, but since it does get pulled from the Web once `/var/lib/crowdsec/data/` is mounted everything is working as designed.

It's technically the case, but we had disabled the MMDBs update in the hub itself due to a potential crash in crowdsec that was fixed in 1.6.3, and we re-enabled it afterwards, so all the next releases should have the latest version from the hub at the moment of the build.
